### DISCUSSION TITLE
HPC Docker Path -- can you point docker invocation at a specific script?

### DISCUSSION THREAD

**Discussion by lstrgar** [nextflow-io/nextflow#2583] [Q&A]
I'm using a nextflow workflow within a shared resource HPC center where docker is not directly accessible. Rather, docker invocation is wrapped in a script with a known location, so if a user wants they will simply replace docker with that path to that script. Sudo privileges also required. 

Is it possible to invoke docker this way with nextflow? When I run the workflow out of the box I see the following error message:

```
Command error:
  /usr/bin/docker-current: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.26/containers/create?name=nxf-F4WjtULxyVlNBD0snlj6Aveo: dial unix /var/run/docker.sock: connect: permission denied.
  See '/usr/bin/docker-current run --help'.
```

So far, I've tried the following approaches -- 

1) In the relevant module file I added `beforeScript "alias docker='/path/to/docker/script'"`
2) In the docker.config file I added `sudo = true`
3) Both 1) and 2) together
4) In the relevant module file I added `beforeScript "alias docker='sudo /path/to/docker/script'"`

Looking at the generated `.command.run` file it's apparent that steps above are being placed in the code, but I'm still not having any success. 

Any suggestions would be so appreciated!

**Comment by pditommaso** [comment]
You may be trying to create a script that invokes your docker and passthrough all arguments ie. 


```
sudo /path/to/your/docker "$@"
```

**Reply by lstrgar** [reply]
You're suggesting appending `"$@"` to the beforeScript?

**Reply by pditommaso** [reply]
No. I'm suggesting to create a script named `docker` with the content above, give execution permission and put in a directory into your `$PATH`
